src/extension/loadLogs.ts

```diff
@@ -33,42 +31,19 @@ export async function loadLogs(uris: Uri[], token?: { isCancellationRequested: b
     logs.forEach(log => Telemetry.sendLogVersion(log.version, log.$schema ?? ''));
     logs.forEach(tryFastUpgradeLog);
 
-    const logsNoUpgrade = [] as Log[];
-    const logsToUpgrade = [] as Log[];
-    const warnUpgradeExtension = logs.some(log => detectUpgrade(log, logsNoUpgrade, logsToUpgrade));
-    const upgrades = logsToUpgrade.length;
-    if (upgrades) {
-        await window.withProgress(
-            { location: ProgressLocation.Notification },
-            async progress => {
-                for (const [i, oldLog] of logsToUpgrade.entries()) {
-                    if (token?.isCancellationRequested) break;
-                    progress.report({
-                        message: `Upgrading ${i + 1} of ${upgrades} log${upgrades === 1 ? '' : 's'}...`,
-                        increment: 1 / upgrades * 100
-                    });
-                    await new Promise(r => setTimeout(r, 0)); // Await otherwise progress does not update. Assumption: await allows the rendering thread to kick in.
-                    const {fsPath} = Uri.parse(oldLog._uri, true);
-                    try {
-                        const tempPath = upgradeLog(fsPath);
-                        const file = fs.readFileSync(tempPath, 'utf8'); // Assume scheme file.
-                        const log = JSON.parse(file) as Log;
-                        log._uri = oldLog._uri;
-                        log._uriUpgraded = Uri.file(tempPath).toString();
-                        logsNoUpgrade.push(log);
-                    } catch (error) {
-                        console.error(error);
-                        window.showErrorMessage(`Failed to upgrade '${fsPath}'`);
-                    }
-                }
-            }
-        );
+    const logsSupported = [] as Log[];
+    const logsNotSupported = [] as Log[];
+    const warnUpgradeExtension = logs.some(log => detectUpgrade(log, logsSupported, logsNotSupported));
+    for (const log of logsNotSupported) {
+        if (token?.isCancellationRequested) break;
+        const {fsPath} = Uri.parse(log._uri, true);
+        window.showWarningMessage(`'${fsPath}' was not loaded. Version '${log.version}' and schema '${log.$schema ?? ''}' is not supported.`);
     }
-    logsNoUpgrade.forEach(log => augmentLog(log, driverlessRules));
+    logsSupported.forEach(log => augmentLog(log, driverlessRules));
     if (warnUpgradeExtension) {
         window.showWarningMessage('Some log versions are newer than this extension.');
     }
-    return logsNoUpgrade;
+    return logsSupported;
 }
 
 export function detectUpgrade(log: Log, logsNoUpgrade: Log[], logsToUpgrade: Log[]) {
```

Comment:
>{ [](start = 84, length = 1)

As long as we have the hood up, I suggest we improve the messaging. Let's have `detectUpgrade` take `supportedLogs`, `oldVersionLogs`, `unrecognizedSchemaLogs`, and `futureVersionLogs`. Place each log in exactly one of those lists. Don't bother with the Boolean return value, which loses the information about which logs were too new. Don't collapse the messaging for version mismatch and schema mismatch.

Once you've processed all the logs and the arrays are filled, then loop over each of `oldVersionLogs`, `unrecognizedSchemaLogs`, and `futureVersionLogs`, and produce the appropiate message:
- `'{fsPath}' was not loaded because this extension does not support the obsolete SARIF version '{log.version}'.`
- `'{fsPath}' was not loaded because this extension does not recognize the schema '{log.$schema}'.`
- `'{fsPath}' was not loaded because its version '{log.version}' is newer than the version '2.1.0' supported by this extension. Consider upgrading to a new version of the extensions.`

Reply:
How frequently do we expect the end user to leverage the finer information? My working experience is that most folks using this don't appreciate schema vs version. This message is essentially an opaque string meant to be forwarded back to us (the developers).

Strictly speaking the end user doesn't even have to open an issue as we have telemtry covering this also.

Reply:
Let's discuss offline when you're back.